src/rdkafka_cgrp.c

```diff
@@ -3241,20 +3546,50 @@ rd_kafka_cgrp_op_serve (rd_kafka_t *rk, rd_kafka_q_t *rkq,
                 break;
 
         case RD_KAFKA_OP_ASSIGN:
-                /* New atomic assignment (payload != NULL),
-                 * or unassignment (payload == NULL) */
-                err = 0;
+                err = RD_KAFKA_RESP_ERR_NO_ERROR;
+                error = NULL;
                 if (rkcg->rkcg_flags & RD_KAFKA_CGRP_F_TERMINATE) {
                         /* Treat all assignments as unassign
                          * when terminating. */
                         rd_kafka_cgrp_unassign(rkcg);
+
                         if (rko->rko_u.assign.partitions)
                                 err = RD_KAFKA_RESP_ERR__DESTROY;
+
+                        error = !err ? NULL
+                                     : rd_kafka_error_new(
+                                                err,
+                                                rd_kafka_err2str(err));
```

Comment:
format string, and provide something more meaningful, e.g., "Consumer is terminating".